modules/sdk/src/main/scala/org/tessellation/sdk/http/routes/ClusterRoutes.scala

```diff
@@ -58,23 +58,18 @@ final case class ClusterRoutes[F[_]: Async](
   private val p2pPublic: HttpRoutes[F] = HttpRoutes.of[F] {
     case req @ POST -> Root / "join" =>
       req.decodeR[JoinRequest] { joinRequest =>
-        req.remoteAddr
-          .flatMap(_.asIpv4)
-          .map(_.toString)
-          .flatMap(Host.fromString)
-          .fold(BadRequest())(host =>
-            joining
-              .joinRequest(collateral.hasCollateral)(joinRequest, host)
-              .flatMap(_ => Ok())
-              .recoverWith {
-                case PeerAlreadyConnected(id, _, _, _) => Conflict(s"Peer id=${id} already connected.")
-                case SessionDoesNotExist               => Conflict("Peer does not have an active session.")
-                case CollateralNotSatisfied            => Conflict("Collateral is not satisfied.")
-                case NodeNotInCluster                  => Conflict("Node is not part of the cluster.")
-                case _                                 => InternalServerError("Unknown error.")
-              }
-          )
-
+        joining
+          .joinRequest(collateral.hasCollateral)(joinRequest, joinRequest.registrationRequest.ip)
+          .flatMap(_ => Ok())
+          .recoverWith {
+            case PeerAlreadyJoinedWithNewerSession(id, _, _, _) => Conflict(s"Peer id=${id} already connected.")
```

Comment:
`id.show` and I'd update the message to match the case class name

Reply:
Fixing